### PR TITLE
feat(docs): improve documentation generation

### PR DESCRIPTION
**Pull request by dpc**
Re #4622

What we'll publish on docs.fedimint.org:

```
nix build .#nightly.ci.workspaceDocExport
xdg-open ./result/share/doc/index.html
```

Handy local version:

```
just docs
```

### CODE REVIEW

File: docs/rustdoc-index.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Fedimint technical reference documentation
+
+<!-- this page is used a landing page for https://docs.fedimint.org/ -->
+
+This is documentation automatically generated from the Fedimint source code.
```
Comment (dpc) [inline docs/rustdoc-index.md:5]: Somewhat of a placeholder for now.

Comment (elsirion) [reply docs/rustdoc-index.md:5]: We should probably outline the structure a bit and link to the most important crates.

**Review by elsirion — DISMISSED** [review]
Really cool!

File: justfile.fedimint.just [hunk 2]
```diff
@@ -172,3 +172,19 @@ fuzz-target-debug TARGET="" CRASH="" *ARGS="--exit_upon_crash":
...
+# Open `cargo doc`-generated documentation
+docs: build-docs
+  #!/usr/bin/env bash
+  source "scripts/_common.sh"
+  main_index_path="${CARGO_BUILD_TARGET_DIR}/doc/index.html"
+  fallback_index_path="${CARGO_BUILD_TARGET_DIR}/doc/fedimint_core/index.html"
+  if [ -e "$main_index_path" ]; then
+    xdg-open "$main_index_path"
+  else
+    xdg-open "$fallback_index_path"
```
Comment (justinmoon) [inline justfile.fedimint.just:189]: Doesn't work on mac. `s/xdg-open/open` does work.

```
/tmp/just-1s8eEV/docs: line 189: xdg-open: command not found
```

**Comment by justinmoon** [comment]
`just docs` works on my mac 🎉

**Review by justinmoon — APPROVED** [review]

**Review by bradleystachurski — COMMENTED** [review]
I needed to add `pkgs.xdg-utils` to `devShell` since I was getting

```
/tmp/nix-shell.qijbQf/just-iYVcB6/docs: line 194: open: command not found
error: Recipe `docs` failed with exit code 127
```

**Comment by dpc** [comment]
Will fallback to one via `nix shell` and print the index file just in case someone wants to know where it is and it didn't open.

**Review by bradleystachurski — APPROVED** [review]

**Review by maan2003 — APPROVED** [review]

**Comment by bradleystachurski** [comment]
Build on linux failed due to known flake (hopefully resolved in https://github.com/fedimint/fedimint/pull/4647).

**Comment by fedimint-backports** [comment]
Git push to origin failed for releases/v0.3 with exitcode 1

**Pull request merged**